lighthouse-core/report/v2/renderer/report-features.js

```diff
@@ -0,0 +1,326 @@
+    const event = new CustomEvent(name, detail ? {detail} : null);
+    this._document.dispatchEvent(event);
+  }
+
+  _setupExportButton() {
+    this.exportButton = this._dom.find('.lh-export__button', this._document);
+    this.exportButton.addEventListener('click', this.onExportButtonClick);
+
+    const dropdown = this._dom.find('.lh-export__dropdown', this._document);
+    dropdown.addEventListener('click', this.onExport);
+  }
+
+  /**
+   * @param {!Element} container Parent element to render the report into.
+   */
+  _setupLogger(container) {
+    const loggerEl = this._dom.createElement('div');
+    loggerEl.id = 'lh-log';
+    container.appendChild(loggerEl);
+    this.logger = new Logger(loggerEl);
+  }
+
+  /**
+   * Handler copy events.
+   * @param {!Event} e
+   */
+  onCopy(e) {
+    // Only handle copy button presses (e.g. ignore the user copying page text).
+    if (this._copyAttempt) {
+      // We want to write our own data to the clipboard, not the user's text selection.
+      e.preventDefault();
+      e.clipboardData.setData('text/plain', JSON.stringify(this.json, null, 2));
+      this.logger.log('Report JSON copied to clipboard');
+    }
+
+    this._copyAttempt = false;
+  }
+
+  /**
+   * Copies the report JSON to the clipboard (if supported by the browser).
+   */
+  onCopyButtonClick() {
+    this._fireEventOn('lh-analytics', this._document, {
+      cmd: 'send',
+      fields: {hitType: 'event', eventCategory: 'report', eventAction: 'copy'}
+    });
+
+    try {
+      if (this._document.queryCommandSupported('copy')) {
+        this._copyAttempt = true;
+
+        // Note: In Safari 10.0.1, execCommand('copy') returns true if there's
+        // a valid text selection on the page. See http://caniuse.com/#feat=clipboard.
+        if (!this._document.execCommand('copy')) {
+          this._copyAttempt = false; // Prevent event handler from seeing this as a copy attempt.
+          this.logger.warn('Your browser does not support copy to clipboard.');
+        }
+      }
+    } catch (/** @type {!Error} */ e) {
+      this._copyAttempt = false;
+      this.logger.log(e.message);
+    }
+  }
+
+  closeExportDropdown() {
+    this.exportButton.classList.remove('active');
+  }
+
+  /**
+   * Click handler for export button.
+   * @param {!Event} e
+   */
+  onExportButtonClick(e) {
+    e.preventDefault();
+    const el = /** @type {!Element} */ (e.target);
+    el.classList.toggle('active');
+    this._document.addEventListener('keydown', this.onKeyDown);
+  }
+
+  /**
+   * Resets the state of page before capturing the page for export.
+   * When the user opens the exported HTML page, certain UI elements should
+   * be in their open state (not opened) and the templates should be refreshly
+   * stamped.
+   */
+  _resetUIState() {
+    this.logger.hide();
+    this.closeExportDropdown();
+    this._dom.findAll('template[data-stamped]', this._document).forEach(t => {
```

Comment:
should this be in this class? It feels like it's a higher-level detail (it's hard to connect to other side of things in `dom.js`)